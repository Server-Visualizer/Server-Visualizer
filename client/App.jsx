import React, { useState, useEffect } from 'react';
import Graphs from './containers/GraphContainer.jsx';
import Requests from './containers/RequestContainer.jsx';
import RequestContext from './contexts/RequestContext.jsx';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      requests: [],
    };    
  }
  componentDidMount() {
    (async () => {
      // Grab requests from our mock server
      const fetched = await fetch('https://sv-mock-data.herokuapp.com/');
      // Parsing the response into a readable JS array
      const requests = await fetched.json();
      this.setState({ requests: requests });
    })();
  }

  render() {
    console.log('context API test', this.context);
    return (
      <div>
        Hello
        <RequestContext.Provider value={this.state.requests}>
          <Graphs />
          <Requests />
        </RequestContext.Provider>
      </div>
    )
  }
}

App.contextType = RequestContext;

// const App = () => {
//   // const [clicks, setClick] = useState([{ wow: 0 }]);
//   const [activities, updateActivities] = useState({});

//   useEffect(() => {
    // fetch('https://sv-mock-data.herokuapp.com/')
    //   // Parsing the response into a readable JS array
    //   .then(res => res.json())
    //   // After converting, store the activities in our state 
    //   .then(res => {
    //     updateActivities(res);
    //     console.log(activities);
    //   })
    //   .catch(err => {
    //     throw err;
    //   }); 
//   }, []);

  // return (
  //   <div>
  //     Hello
  //     <Graphs />
  //     <Requests />
  //     {/* <button type="button" onClick={()=>{ setClick({ wow: clicks.wow + 1 }) }}>Test</button> */}
  //   </div>
  // )
// }


export default App;