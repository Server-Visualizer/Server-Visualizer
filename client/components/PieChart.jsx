import React from 'react';
import RequestContext from '../contexts/RequestContext.jsx';

class PieChart extends React.Component {
  render() {
    console.log('context in piechart', this.context);
    return (
      <div>
      I'm a pie chart
      </div>
    )
  }
}

PieChart.contextType = RequestContext;
export default PieChart;