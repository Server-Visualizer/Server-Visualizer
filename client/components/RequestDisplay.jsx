import React from 'react';


class RequestDisplay extends React.Component {
  render() {
    const { method, reqTime, elapsedTime, status } = this.props.reqInfo;
    const parsedReqTime = new Date(reqTime).toLocaleTimeString();
    const parsedElapsedTime = elapsedTime.toPrecision(1);
    return (
      <div className="request">
        <p><strong>Method:</strong> {method}</p>
        <p><strong>Status:</strong> {status}</p>
        <p><strong>Request Time:</strong> {parsedReqTime}</p>
        <p><strong>Processing Time:</strong> {parsedElapsedTime}ms</p>
      </div>
    )
  }
}

export default RequestDisplay;