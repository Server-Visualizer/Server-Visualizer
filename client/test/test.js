import React from 'react'; 
// Import testing 
import Enzyme, { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16'; 
import BabelPolyfill from 'babel-polyfill'; 
// Main app 
import App from '../App.jsx'; 
// Containers
import Graphs from '../containers/GraphContainer.jsx';
import Requests from '../containers/RequestContainer.jsx';
// Components 
import RequestDisplay from '../components/RequestDisplay.jsx';
import PieChart from '../components/PieChart.jsx'; 
// Import mock fetch data
import fetch from './__mocks__/fetch'; 
global.fetch = fetch; 
// Configure the adapter 
Enzyme.configure({ adapter: new Adapter() }); 

// App should render its components without crashing 
describe('App component', () => {
  // test the initial layout
    it('should render initial layout', () => {
      const component = shallow(<App />); 
      expect(component.getElements()).toMatchSnapshot(); 
    }); 
}); 
// Test our fetch 
describe('App component', () => {
  describe('When rendered', () => {
    it('should fetch the requests', () => {
      // fetch using the mock fetch method
      const fetchSpy = jest.spyOn(window, 'fetch');
      const wrapper = shallow(<App />); 
      expect(fetchSpy).toBeCalled(); 
    })
  })
}); 

// GraphContainer should render 
describe('Graph Container component', () => {
  // test the initial layout
    it('should render initial layout', () => {
      const component = shallow(<Graphs />); 
      expect(component.getElements()).toMatchSnapshot(); 
    }); 
    it('renders correctly', () => {
      // test the shallow copy
        const wrapper = shallow(<Graphs context={[{test: test}]}/>); 
        expect(wrapper).toMatchSnapshot(); 
    }); 
}); 

// Request Display component testing
describe('Request Display component', () => {
  // test the initial layout
  it('should render initial layout', () => {
      const component = shallow(<RequestDisplay reqInfo={{ method: "GET", reqTime: 10, elapsedTime: 10, status: 200}}/>); 
      expect(component.getElements()).toMatchSnapshot(); 
  }); 
  // It should render 
    it('renders correctly', () => {
        const wrapper = shallow(<RequestDisplay reqInfo={{ method: "GET", reqTime: 10, elapsedTime: 10, status: 200}} />); 
        expect(wrapper).toMatchSnapshot(); 
    }); 
}); 

