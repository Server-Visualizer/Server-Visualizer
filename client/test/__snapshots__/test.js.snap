// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render initial layout 1`] = `
Array [
  <div
    id="app"
  >
    <Context.Provider
      value={Array []}
    />
  </div>,
]
`;

exports[`Graph Container component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Graph Container component should render initial layout 1`] = `
Array [
  <div
    id="graph-container"
  >
    <PieChart
      data={Object {}}
    />
    <CodeChart
      data={Object {}}
    />
  </div>,
]
`;

exports[`Request Display component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Request Display component should render initial layout 1`] = `
Array [
  <div
    className="request"
  >
    <p>
      <strong>
        Method:
      </strong>
       
      GET
    </p>
    <p>
      <strong>
        Status:
      </strong>
       
      200
    </p>
    <p>
      <strong>
        Request Time:
      </strong>
       
      7:00:00 PM
    </p>
    <p>
      <strong>
        Processing Time:
      </strong>
       
      1e+1
      ms
    </p>
  </div>,
]
`;
